api/import.go

```diff
@@ -52,3 +56,31 @@ func ImportChannel(channel *model.Channel) *model.Channel {
 		return sc
 	}
 }
+
+func ImportFile(file io.Reader, teamId string, channelId string, userId string, fileName string) (string, error) {
+	buf := bytes.NewBuffer(nil)
+	io.Copy(buf, file)
+
+	uid := model.NewId()
+
+	imageNameList := []string{}
+	imageDataList := [][]byte{}
+
+	if model.IsFileExtImage(filepath.Ext(fileName)) {
+		imageNameList = append(imageNameList, uid+"/"+fileName)
+		imageDataList = append(imageDataList, buf.Bytes())
+	}
+
+	path := "teams/" + teamId + "/channels/" + channelId + "/users/" + userId + "/" + uid + "/" + fileName
+
+	if err := WriteFile(buf.Bytes(), path); err != nil {
+		return "", err
+	}
+
+	encName := utils.UrlEncode(fileName)
+	fileUrl := "/" + channelId + "/" + userId + "/" + uid + "/" + encName
+
+	go handleImages(imageNameList, imageDataList, teamId, channelId, userId)
+
+	return fileUrl, nil
+}
```

Comment:
Latest version is now compatible with the new files API. Thanks for the pointers.